src/backend/optimizer/plan/planshare.c

```diff
@@ -211,6 +211,14 @@ prepare_plan_for_sharing(PlannerInfo *root, Plan *common)
 		Material *m = make_material(common);
 		shared = (Plan *) m;
 
+		if (!shared->targetlist &&
+			 IsA(common, ModifyTable))
+		{
+			ModifyTable *mt = (ModifyTable *)common;
+			shared->targetlist = copyObject(linitial(mt->returningLists));
+		}
+		Assert(shared->targetlist);
```

Comment:
Do we need that assert? If Material's targetList will be empty we will get the same error as in current ticket. It's unnecessary.

Reply:
does this assert describe the problem better? because the original one says that the planner requires projections for the node that does not support it, whereas the problem is not in this.